lrpar/src/lib/ctbuilder.rs

```diff
@@ -150,6 +197,69 @@ where
         self
     }
 
+    /// If set to true, `process_file_in_src` will return an error if the given grammar contains
+    /// any Shift/Reduce or Reduce/Reduce conflicts. Defaults to `true`.
+    pub fn error_on_conflicts(mut self, b: bool) -> Self {
+        self.error_on_conflicts = b;
+        self
+    }
+
+    pub fn conflicts(&self) -> Option<&Conflicts<StorageT>> {
+        if let Some((_, _, s)) = &self.error_data {
+            s.conflicts()
+        }
+        else {
+            None
+        }
+    }
+
+    pub fn grm(&self) -> Option<&YaccGrammar<StorageT>> {
+        if let Some((g, _, _)) = &self.error_data {
+            Some(&g)
+        }
+        else {
+            None
+        }
+    }
+
+    pub fn sgraph(&self) -> Option<&StateGraph<StorageT>> {
+        if let Some((_, sg, _)) = &self.error_data {
+            Some(&sg)
+        }
+        else {
+            None
+        }
+    }
+
+    pub fn stable(&self) -> Option<&StateTable<StorageT>> {
+        if let Some((_, _, st)) = &self.error_data {
+            Some(&st)
+        }
+        else {
+            None
+        }
+    }
+
+    /// Returns a pretty-printed version of the graph.
+    pub fn pp_sgraph(&self) -> String {
```

Comment:
Can't someone do `sgraph().pp_core_states(&grm)`? I'm not sure we need this function.

Reply:
They can, but then they need to retrieve `grm` first from the `Conflicts` struct as it's needed as an argument. I thought this was more user friendly.